types/plugin.ts

```diff
@@ -37,3 +43,302 @@ export const Plugins: Record<PluginID, Plugin> = {
+            content[key] = {
+              schema: contentObject[key].schema,
+            };
+          }
+          pluginApiOperation.responses[key].content = content;
+        }
+      }
+
+      pluginApiOperationList[operationObject.operationId] = pluginApiOperation;
+    }
+  }
+
+  return pluginApiOperationList;
+}
+
+export function getOpenAIFunctionFromPluginApiOperation(
+  pluginApiOperation: PluginApiOperation,
+): OpenAIFunction {
+  const {
+    operationId: id,
+    summary,
+    parameters,
+    requestBody,
+  } = pluginApiOperation;
+
+  // Build the properties object for OpenAIFunction
+  const properties: { [key: string]: {} } = {};
+  if (parameters) {
+    for (const parameter of parameters) {
+      properties[parameter.name] = parameter.schema;
+    }
+  }
+
+  // Build the required array for OpenAIFunction
+  const required =
+    parameters
+      ?.filter((parameter) => parameter.required)
+      .map((parameter) => parameter.name) || [];
+
+  if (requestBody) {
+    for (const key in requestBody.content) {
+      for (const propertyKey in requestBody.content[key].schema.properties) {
+        properties[propertyKey] = {
+          type: requestBody.content[key].schema.properties[propertyKey].type,
+          description:
+            requestBody.content[key].schema.properties[propertyKey]
+              .description || '',
+        };
+        if (requestBody.content[key].schema.properties[propertyKey].required) {
+          required.push(propertyKey);
+        }
+      }
+    }
+  }
+
+  // Create OpenAIFunction object
+  const openAIFunction: OpenAIFunction = {
+    name: id,
+    description: summary || '',
+    parameters: {
+      type: 'object',
+      properties: properties,
+      required: required,
+    },
+  };
+
+  return openAIFunction;
+}
+
+export const runPluginApiOperation = async (
+  operation: PluginApiOperation,
+  args: string,
+) => {
+  let query: { [key: string]: string } = {};
+  if (operation.parameters) {
+    for (const parameter of operation.parameters) {
+      if (parameter.in === 'path' && args.includes(parameter.name)) {
+        const newPath = JSON.parse(args)[parameter.name];
+        operation.apiPath = operation.apiPath.replace(
+          `{${parameter.name}}`,
+          newPath,
+        );
+      }
+      if (parameter.in === 'query' && args.includes(parameter.name)) {
+        const newQuery = JSON.parse(args)[parameter.name];
+        query[parameter.name] = newQuery;
+      }
+    }
+  }
+
+  let body: any = {};
+  if (operation.requestBody) {
+    for (const key in operation.requestBody.content) {
+      for (const propertyKey in operation.requestBody.content[key].schema
+        .properties) {
+        if (args.includes(propertyKey)) {
+          body[propertyKey] = JSON.parse(args)[propertyKey];
+        }
+      }
+    }
+    body = JSON.stringify(body);
+  } else {
+    body = undefined;
+  }
+
+  let url = operation.serverUrl + operation.apiPath;
+  if (query) {
```

Comment:
This check is always truthy, because of `let query = {}` initialization above.
Either it should be initialized inside the `if (parameter.in === 'query')` block or you could also simply check if there are any keys in the object:
```
if (Object.keys(query).length > 0) {
    url = url + '?' + new URLSearchParams(query).toString();
  }
```